### PR TITLE
Web: remove s3 bucket option when creating/editing aws oidc integration

### PR DESCRIPTION
**Pull request by kimlisa**
part of https://github.com/gravitational/teleport/issues/44053

recommend reviewing by commit

🛑 requires e2e testing before merging 

- Removes s3 fields upon creating new aws oidc integration
- Removes s3 fields when editing existing integration
- Existing integrations with s3 buckets already configured
  - renders a warning that users should reconfigure
  - does not allow user to edit s3 fields (read only)

Editing with existing s3 bucket fields
image

After pressing `reconfigure` button (the info is shown only for s3 bucket user)
image

Editing without s3 bucket fields
image

changelog: Remove deprecated S3 bucket option when creating or editing AWS OIDC integration in the web UI

### CODE REVIEW

File: web/packages/teleport/src/Integrations/EditAwsOidcIntegrationDialog.tsx [hunk 1]
```diff
@@ -146,9 +146,8 @@ export function EditAwsOidcIntegrationDialog(props: Props) {
                     s3Prefix={s3Prefix}
                   />
                   <OutlineWarn>
-                    Next reconfiguration will remove this integrations S3 bucket
-                    automatically. Storing the OpenID configuration in a S3
-                    bucket is deprecated.
+                    Reconfiguring is suggested to remove the deprecated S3
+                    fields from this integration.
```
Comment (avatus) [inline web/packages/teleport/src/Integrations/EditAwsOidcIntegrationDialog.tsx:150]: I'm not 100% sure I understand what this is saying. I'm sure there is even better copy, but I'll try a suggestion and you can decide what you like
"We suggest reconfiguring this integration to remove the deprecated s3 fields"

File: web/packages/teleport/src/Integrations/Enroll/AwsOidc/S3BucketConfiguration.tsx [hunk 2]
```diff
@@ -33,8 +33,7 @@ export function S3BucketConfiguration({
       <Flex alignItems="center" gap={1}>
         <Text>Amazon S3 Location</Text>
         <ToolTipInfo sticky={true} kind="warning">
-          Deprecated. We suggest to reconfigure this integration which will
-          remove the S3 bucket for you.
+          Storing the OpenID configuration in a S3 bucket is now deprecated.
```
Comment (avatus) [inline web/packages/teleport/src/Integrations/Enroll/AwsOidc/S3BucketConfiguration.tsx:36]:
```suggestion
          Storing the OpenID configuration in an S3 bucket is now deprecated.
```

**Review by avatus — APPROVED** [review]

File: web/packages/teleport/src/Integrations/EditAwsOidcIntegrationDialog.tsx [hunk 3]
```diff
@@ -151,16 +139,20 @@ export function EditAwsOidcIntegrationDialog(props: Props) {
...
-              />
+              {showReadonlyS3Fields && !scriptUrl && (
+                <>
+                  <S3BucketConfiguration
+                    s3Bucket={s3Bucket}
+                    s3Prefix={s3Prefix}
+                  />
+                  <OutlineWarn>
+                    We suggest reconfiguring this integration to remove the
+                    deprecated s3 fields.
```
Comment (gzdunek) [inline web/packages/teleport/src/Integrations/EditAwsOidcIntegrationDialog.tsx:150]:
```suggestion
                    deprecated S3 fields.
```

File: web/packages/teleport/src/Integrations/Enroll/AwsOidc/S3BucketConfiguration.tsx [hunk 4]
```diff
@@ -21,62 +21,35 @@ import { Text, Flex } from 'design';
...
   return (
     <>
       <Flex alignItems="center" gap={1}>
         <Text>Amazon S3 Location</Text>
-        <ToolTipInfo>
-          Teleport will create and use Amazon S3 Bucket as this integration's
-          issuer and will publicly host two files: one for the OpenID
-          configuration and another one for the public key.
+        <ToolTipInfo sticky={true} kind="warning">
+          Deprecated.
```
Comment (gzdunek) [inline web/packages/teleport/src/Integrations/Enroll/AwsOidc/S3BucketConfiguration.tsx:36]: Should we also say what to use instead?

**Review by gzdunek — APPROVED** [review]

**Pull request merged**